azure/services/aso/aso.go

```diff
@@ -156,7 +149,11 @@ func (r *reconciler[T]) CreateOrUpdateResource(ctx context.Context, spec azure.A
 	}
 
 	// Construct parameters using the resource spec and information from the existing resource, if there is one.
-	parameters, err := PatchedParameters(ctx, r.Scheme(), spec, existing.DeepCopy())
+	var existingCopy T
+	if resourceExists {
+		existingCopy = existing.DeepCopyObject().(T)
```

Comment:
This rules, I had to read it a couple times before I understood.

Reply:
It's all thanks to @nojnhuh! Actually, why is this deep copy needed? I don't think `PatchedParameters` modifies `existing` in any way.